h2o-genmodel-extensions/jgrapht/src/main/java/hex/genmodel/tools/JgraphtPrintMojo.java

```diff
@@ -0,0 +1,227 @@
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.Map;
+
+public class JgraphtPrintMojo extends PrintMojo implements MojoPrinter {
+
+    public static void main(String[] args) {
+        MojoPrinter mojoPrinter;
+        if ("png".equals(getFormat(args))){
+            mojoPrinter = new JgraphtPrintMojo();
+        } else {
+            mojoPrinter = new PrintMojo();
+        }
+        mojoPrinter.parseArgs(args);
+
+        // Run the main program
+        try {
+            mojoPrinter.run();
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.exit(2);
+        }
+        
+        // Success
+        System.exit(0);
+    }
+    
+    @Override
+    protected void validateArgs() {
+        if (genModel == null) {
+            System.out.println("ERROR: Must specify -i");
+            usage();
+        }
+    }
+
+    @Override
+    public void run() throws Exception {
+        validateArgs();
+     
+        if (genModel instanceof TreeBackedMojoModel){
+            TreeBackedMojoModel treeBackedModel = (TreeBackedMojoModel) genModel;
+            ConvertTreeOptions options = new ConvertTreeOptions().withTreeConsistencyCheckEnabled();
+            final SharedTreeGraph g = treeBackedModel.convert(treeToPrint, null, options);
+            printPng(g);
+        }
+        else {
+            System.out.println("ERROR: Unknown MOJO type");
+            System.exit(1);
+        }
+    }
+
+    private void printPng(SharedTreeGraph trees) throws IOException, ImportException {
+        Path outputDirectoryPath = Paths.get(outputFileName);
+        int numberOfTrees = trees.subgraphArray.size();
+        if (Files.exists(outputDirectoryPath)) {
+            Files.walk(outputDirectoryPath)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(File::delete);
+        }
+        if (numberOfTrees > 1) { 
+            if (outputFileName == null) {
+                outputFileName = Paths.get("").toString();
+            }
+            Files.createDirectory(outputDirectoryPath);
+        }
+        for (SharedTreeSubgraph tree : trees.subgraphArray) {
+            Path dotSourceFilePath = Files.createTempFile("", tmpOutputFileName);
+            try (FileOutputStream fosTemp = new FileOutputStream(dotSourceFilePath.toFile()); PrintStream osTemp = new PrintStream(fosTemp)) {
+                tree.printDot(osTemp, maxLevelsToPrintPerEdge, detail, optionalTitle, pTreeOptions, true);
+                generateOutputPng(dotSourceFilePath, getPngName(numberOfTrees, tree.name));
+            }
+            Files.delete(dotSourceFilePath);
+        }
+
+    }
+
+    private void generateOutputPng(Path dotSourceFilePath, String treeName) throws ImportException, IOException {
```

Comment:
confusing argName, its actualy output file path

Reply:
it is not / it is a path to .dot|.gv file from which we generate .png graph